lib/ansible/modules/cloud/docker/docker_node_facts.py

```diff
@@ -52,31 +64,39 @@
 '''
 
 EXAMPLES = '''
+- name: Get info on all nodes
+  docker_node_facts:
+  register: result
+
 - name: Get info on node
   docker_node_facts:
     name: mynode
   register: result
 
+- name: Get info on list of nodes
+  docker_node_facts:
+    name:
+      - mynode1
+      - mynode2
+  register: result
+
+- name: Get info on host if it is Swarm Manager
+  docker_node_facts:
+    self: true
+  register: result
 '''
 
 RETURN = '''
-exists:
+nodes_facts:
     description:
-      - Returns whether the node exists in docker swarm cluster.
-    type: bool
+      - Facts representing the current state of the nodes. Matches the C(docker node inspect) output.
+      - Will be C(none) if node does not exist.
+      - Will contain multiple entries if more than one node provided in I(name).
+      - If I(name) contains list of nodes, the output will contain information only about registered nodes.
```

Comment:
registered nodes == nodes that exist?

Reply:
Not exactly, it may need better phrasing. If a node leaves the swarm via `docker swarm leave` it will still exist on the `docker node ls` output on the manager, and you can get the facts about it, just its status will be `Down`. So technically the node is not part of the swarm as it left (same for the node that lost the connectivity) but is still registered at swarm manager until `docker node rm` is not executed. Maybe a better description would be:

```
- If I(name) contains a list of nodes, the output will contain information on all nodes registered at the swarm, including nodes that left the swarm but hasn't been removed from the cluster on managers hosts and nodes that are unreachable.
```

Reply:
Let me know if the updated description in the latest commits looks better